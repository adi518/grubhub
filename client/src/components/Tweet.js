import React from 'react';
import styled from 'styled-components';

import Likes from './Likes';
import Avatar from './Avatar';

const StyledTweet = styled.div`
  width: 100%;
  display: flex;
  padding: 1rem;
  color: #ffffff;
  flex-shrink: 0;
  min-height: 5rem;
  border-radius: 1rem;
  margin-bottom: 1rem;
  flex-direction: column;
  box-sizing: border-box;
  border: 1px solid #38444d;
  transition: background-color 0.5s, border-color 0.5s;

  &:hover {
    border-color: #1b95e0;
    background-color: #1b95e0;
  }

  &:first-child {
    margin-top: 1rem;
  }
`;

const FlexRow = styled.div`
  display: flex;
`;

const Text = styled.div`
  margin-top: 0.25rem;
  word-break: break-all;
`;

const Tweet = ({ user, text, favorite_count }) => (
  <StyledTweet>
    <FlexRow>
      <Avatar user={user} />
      <div>
        <b>@{user.name}</b>
        <Text>{text}</Text>
      </div>
    </FlexRow>
    <Likes count={favorite_count} />
  </StyledTweet>
);

export default Tweet;
