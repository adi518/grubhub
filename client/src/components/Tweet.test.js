import React from 'react';
import Tweet from './Tweet';
import { render } from '@testing-library/react';

test('Tweet component', () => {
  const { asFragment } = render(
    <Tweet
      user={{
        name: 'Driver Man',
        profile_image_url:
          'http://pbs.twimg.com/profile_images/1201972977396379648/5ZQCnQOa_normal.jpg'
      }}
      text="Block You - Driver Man Vlog 12 https://t.co/e8S0XNCqgq #coronavirus #blockyou"
      favorite_count={2}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
