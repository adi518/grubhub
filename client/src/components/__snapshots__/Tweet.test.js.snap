// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet component 1`] = `
<DocumentFragment>
  <div
    class="sc-htpNat bqsmbJ"
  >
    <div
      class="sc-bxivhb cDFCAd"
    >
      <div
        class="sc-bwzfXH fwjInP"
      >
        <img
          alt="Driver Man"
          src="http://pbs.twimg.com/profile_images/1201972977396379648/5ZQCnQOa_normal.jpg"
        />
      </div>
      <div>
        <b>
          @Driver Man
        </b>
        <div
          class="sc-ifAKCX iZRSX"
        >
          Block You - Driver Man Vlog 12 https://t.co/e8S0XNCqgq #coronavirus #blockyou
        </div>
      </div>
    </div>
    <div
      class="sc-bdVaJa ipDYfv"
    >
       2 üëç 
    </div>
  </div>
</DocumentFragment>
`;
